File: LAB_1/LAB_1_Java/src/ThreadBreaker.java
import java.util.Arrays;
public class ThreadBreaker implements Runnable
{
    private Counter[] _conters;

    public ThreadBreaker(Counter[] counters)
    {
        _conters = counters;
        Arrays.sort(_conters);
    }

    @Override
    public void run()
    {
        long currentWaitedTime = 0;

        for (int i = 0; i<_conters.length;i++){
            long waitingTime = _conters[i].getTimeOfWork() - currentWaitedTime;
            try {
                Thread.sleep(waitingTime);
                currentWaitedTime += waitingTime;
                _conters[i].Switch();
            } catch (InterruptedException e) {
                throw new RuntimeException(e);
            }
        }
    }
}

File: LAB_1/LAB_1_Csharp/LAB_1_Csharp/ThreadBreaker.cs
using System;
using System.Threading;

namespace LAB_1_Csharp
{
    public class ThreadBreaker
    {
        private Counter[] _counters;
        public ThreadBreaker(Counter[] counters)
        {
            _counters = counters;
            Array.Sort(counters);
        }

        public void Stop()
        {
            int currentWaitedTime = 0;

            for (int i = 0; i < _counters.Length; i++)
            {
                int waitingTime = _counters[i].TimeOfWorking - currentWaitedTime;
                Thread.Sleep(waitingTime);
                currentWaitedTime += waitingTime;
                _counters[i].Switch();
            }
        }
    }
}
